utility/get_ip_addrs.py

```diff
@@ -0,0 +1,54 @@
+""" Look at Ceph storage pods, return ip-addresses """
+import subprocess
+import shlex
+from ocs.defaults import ROOK_CLUSTER_NAMESPACE
+from ocs import api_client as ac
+
+
+def get_ip_addrs_for_ceph_type(pod_type, namespace=ROOK_CLUSTER_NAMESPACE):
+    """
+    Get the ip address for all pods of a given type.
+
+    Args:
+        pod_type (str): pod type (ceph-mon, ceph-mgr, ceph-osd)
+        namespace (str): namespace in -n field of oc commands
+                         Defaults to ROOK_CLUSTER_NAMESPACE
+
+    Returns:
+        list: List of ip_addresses for all pods whose name contains
+              the text in pod type
+
+    """
+    ipaddrs = get_ip_addrs(namespace)
+    return [ipaddrs[cpod] for cpod in ipaddrs if pod_type in cpod]
+
+
+def get_ip_addrs(namespace='openshift-storage'):
+    """
+    Get the ip addresses of all pods in the namespaces.
+
+    Args:
+        namespace (str): namespace in -n field of oc commands
+                         Defaults to ROOK_CLUSTER_NAMESPACE
+
+    Returns:
+        dictionary of ip_address values indexed by pod_name.
+
+    """
+    ret_value = {}
+    client = ac.get_api_client("OCRESTClient")
+    for pod in client.get_pods(namespace=namespace):
+        oc_cmd = f"oc -n {namespace} describe pod {pod}"
+        process = subprocess.Popen(shlex.split(oc_cmd), stdout=subprocess.PIPE)
+        output, error = process.communicate()
```

Comment:
lines 41 - 43:
There's already implementation for this - please use it

Reply:
When I try use the ocs implementation, I get: 

 File "/home/ec2-user/ocs-ci/utility/get_ip_addrs.py", line 46, in get_ip_addrs
    OCP.exec_oc_cmd(oc_cmd)
  File "/home/ec2-user/ocs-ci/ocs/ocp.py", line 66, in exec_oc_cmd
    return munchify(yaml.safe_load(out))
  File "/home/ec2-user/venv/lib/python3.7/site-packages/yaml/__init__.py", line 162, in safe_load
    return load(stream, SafeLoader)
  File "/home/ec2-user/venv/lib/python3.7/site-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
  File "/home/ec2-user/venv/lib/python3.7/site-packages/yaml/constructor.py", line 41, in get_single_data
    node = self.get_single_node()
  File "/home/ec2-user/venv/lib/python3.7/site-packages/yaml/composer.py", line 36, in get_single_node
    document = self.compose_document()
  File "/home/ec2-user/venv/lib/python3.7/site-packages/yaml/composer.py", line 55, in compose_document
    node = self.compose_node(None, None)
  File "/home/ec2-user/venv/lib/python3.7/site-packages/yaml/composer.py", line 84, in compose_node
    node = self.compose_mapping_node(anchor)
  File "/home/ec2-user/venv/lib/python3.7/site-packages/yaml/composer.py", line 127, in compose_mapping_node
    while not self.check_event(MappingEndEvent):
  File "/home/ec2-user/venv/lib/python3.7/site-packages/yaml/parser.py", line 98, in check_event
    self.current_event = self.state()
  File "/home/ec2-user/venv/lib/python3.7/site-packages/yaml/parser.py", line 428, in parse_block_mapping_key
    if self.check_token(KeyToken):
  File "/home/ec2-user/venv/lib/python3.7/site-packages/yaml/scanner.py", line 116, in check_token
    self.fetch_more_tokens()
  File "/home/ec2-user/venv/lib/python3.7/site-packages/yaml/scanner.py", line 223, in fetch_more_tokens
    return self.fetch_value()
  File "/home/ec2-user/venv/lib/python3.7/site-packages/yaml/scanner.py", line 579, in fetch_value
    self.get_mark())
yaml.scanner.ScannerError: mapping values are not allowed here
  in "<unicode string>", line 10, column 37:
    Annotations:        openshift.io/scc: rook-ceph

I am still investigating

Reply:
I have opened an issue for this problem.  I think that some text returned by the describe pod command is conflusing the api_client.